kzg_prover/src/circuits/utils.rs

```diff
@@ -192,82 +185,74 @@ pub fn verify_grand_sum_openings<const N_CURRENCIES: usize>(
         }
     }
 
-    let mut verification_results = Vec::<bool>::new();
-    let mut constant_terms = Vec::<BigUint>::new();
-
-    for (i, advice_commitment) in advice_commitments.iter().enumerate() {
-        let (verified, constant_term) = verify_opening::<
-            KZGCommitmentScheme<Bn256>,
-            VerifierSHPLONK<'_, Bn256>,
-            Challenge255<G1Affine>,
-            Blake2bRead<_, _, Challenge255<_>>,
-            AccumulatorStrategy<_>,
-        >(
-            params,
-            &challenge_opening_proofs[i],
-            Fp::zero(),
-            *advice_commitment,
-        );
-        verification_results.push(verified);
-
-        if verified {
-            constant_terms.push(fp_to_big_uint(constant_term * Fp::from(polynomial_degree)));
-        } else {
-            constant_terms.push(BigUint::from(0u8));
-        }
-    }
-    (verification_results, constant_terms)
+    let (verified, constant_terms) = verify_opening::<
+        KZGCommitmentScheme<Bn256>,
+        VerifierSHPLONK<'_, Bn256>,
+        Challenge255<G1Affine>,
+        Blake2bRead<_, _, Challenge255<_>>,
+        AccumulatorStrategy<_>,
+        N_CURRENCIES,
+    >(
+        params,
+        &challenge_opening_multi_proof,
+        Fp::zero(),
+        &advice_commitments,
+    );
+
+    (
+        verified,
+        constant_terms
+            .iter()
+            .map(|eval| fp_to_big_uint(eval * Fp::from(polynomial_degree)))
+            .collect(),
+    )
 }
 
 pub fn verify_user_inclusion<const N_CURRENCIES: usize>(
     params: &ParamsKZG<Bn256>,
     zk_snark_proof: &[u8],
-    balance_opening_proofs: Vec<Vec<u8>>,
+    balance_opening_multi_proof: &[u8],
     balance_column_range: Range<usize>,
     omega: Fp,
     user_index: u16,
-) -> (Vec<bool>, Vec<BigUint>) {
+) -> (bool, Vec<BigUint>) {
```

Comment:
Maybe it's more intuitive to separate userID and balances when returned from `verify_user_inclusion`

Reply:
Not sure about that because all the KZG commitments are treated in the same way here. The caller knows what columns are balances and what is the user ID, and the caller could pass these column indices into the function, but the only thing the function would do with them is using them to split the output for the caller (who can do it themselves because of knowing what columns they are).

Reply:
I understand your explanation and makes sense to me!